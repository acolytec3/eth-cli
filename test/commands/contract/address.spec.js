const ContractAddressCommand = require('../../../src/commands/contract/address')

describe('contract:address', () => {
  let stdoutResult

  beforeEach(() => {
    stdoutResult = []
    jest
      .spyOn(process.stdout, 'write')
      .mockImplementation(val => stdoutResult.push(require('strip-ansi')(val.toString('utf8'))))
  })

  afterEach(() => jest.restoreAllMocks())

  it(`Should run 'contract:address' with empty args and flags and throw an error.`, async () => {
    await expect(ContractAddressCommand.run()).rejects.toThrow()
  })

  it(`Should run 'contract:address' and throw an error.`, async () => {
    await expect(ContractAddressCommand.run([])).rejects.toThrow('address is required')
  })

  it(`Should run 'contract:address --help' and throw an error.`, async () => {
    await expect(ContractAddressCommand.run(['--help'])).rejects.toThrow('EEXIT: 0')
  })

  it(`Should run 'contract:address --bar' and be get some value.`, async () => {
    await ContractAddressCommand.run(['--bar'])
    expect(stdoutResult).toMatchSnapshot()
  })

  it(`Should run 'contract:address 0x92970dbD5C0Ee6b439422bFd7cD71e1DDA921A03' and get success.`, async () => {
    await ContractAddressCommand.run(['0x92970dbD5C0Ee6b439422bFd7cD71e1DDA921A03'])
    expect(stdoutResult).toMatchSnapshot()
  })
})
