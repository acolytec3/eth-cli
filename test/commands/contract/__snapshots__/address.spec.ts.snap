// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contract:address Should run 'contract:address --bar' and be get some value. 1`] = `
Array [
  "0xe01c10fd900939d1eab56ee373ea5e2bd4e2cfb3
",
]
`;

exports[`contract:address Should run 'contract:address 0x92970dbD5C0Ee6b439422bFd7cD71e1DDA921A03' and get success. 1`] = `
Array [
  "0x68ba6833b22f414e97a7aa23908428163be96e9b
",
]
`;
